### DISCUSSION TITLE
Airflow 2.9.1 : AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.

### DISCUSSION THREAD

**Discussion by michaelritsema** [apache/airflow#40113] [Q&A]
### Apache Airflow version

2.9.1

### What happened?

related: #20603 

I installed from PIP and in development mode on a mac. I'm experiencing this same issue when calling get_context()

I'm new to airflow but it seems like there are still references to this in the code:

class _BasePythonVirtualenvOperator(PythonOperator, metaclass=ABCMeta):
    BASE_SERIALIZABLE_CONTEXT_KEYS = {
        "ds",
        "ds_nodash",
        "expanded_ti_count",
        "inlets",
        "map_index_template",
        "next_ds",
        "next_ds_nodash",
        "outlets",
        "prev_ds",
        "prev_ds_nodash",
        "run_id",
        "task_instance_key_str",
        "test_mode",
        "tomorrow_ds",
        "tomorrow_ds_nodash",
        "ts",
        "ts_nodash",
        "ts_nodash_with_tz",
        "yesterday_ds",
        "yesterday_ds_nodash",
    }

expanded log:

/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
/Users/mr/airflow/lib/python3.11/site-packages/airflow/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.

### How to reproduce

any call to get_context()

### Operating System

Sonoma 14.5

### Versions of Apache Airflow Providers

airflow-clickhouse-plugin==1.3.0
apache-airflow==2.9.1
apache-airflow-providers-common-io==1.3.2
apache-airflow-providers-common-sql==1.14.0
apache-airflow-providers-fab==1.1.1
apache-airflow-providers-ftp==3.9.1
apache-airflow-providers-http==4.11.1
apache-airflow-providers-imap==3.6.1
apache-airflow-providers-smtp==1.7.1
apache-airflow-providers-sqlite==3.8.1

### Deployment

Astronomer

### Deployment details

local pip

### Anything else?

,

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by shahar1** [comment]
Welcome, and thank you for your post!
The issue that you're trying to address is not clear:
1. By `get_context()` you mean `get_current_context()`? Do you use it according to the [docs](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/python/index.html)?
2. You specify `Astronomer` as the deployment environment. Do you use their framework or install Airflow otherwise?
3. Could you please provide an example of a DAG that provides this warning? Does this warning affect the functionality of the DAG?

As it currently seems more of a troubleshooting than a problem that needs fixing, I'm converting this into a Q&A discussion.

**Comment by michaelritsema** [comment]
I don't use astronomer. get_current_context() throws this warning on a fresh install from pip on mac in the versions I specified. I posted a reference to deprecated variables in the code. 
```
  def feeds():
      @task(map_index_template="{{ task_index }}")
      def load_staging_task(feed_type,params=None):
          client = ClickHouseHook() 
          table_name = feed_config[feed_type]['table_name']
          context = get_current_context()
          context["task_index"] = "load_staging_task_" + feed_type
```
There is no need to keep this ticket open as QA it doesn't matter to me if you fix it

**Comment by elehcim** [comment]
Hello, I'm experiencing the same issue here, with 2.9.2. 
Also for me the issue arises when  assigning a `map_index_template` inside a task. If I do that, the deprecation warnings are fired. If don't, no deprecation warnings are printed.

**Reply by shahar1** [reply]
> Hello, I'm experiencing the same issue here, with 2.9.2. Also for me the issue arises when assigning a `map_index_template` inside a task. If I do that, the deprecation warnings are fired. If don't, no deprecation warnings are printed.

Could you please provide a minimal example to reproduce the issue?
If you could, you may recreate it as a new issue (preferrably with screenshots attached).

**Comment by foldvari** [comment]
@shahar1 

Minimal code to reproduce the problem:

```
from datetime import datetime

from airflow import DAG
from airflow.decorators import task
from airflow.operators.python import get_current_context

with DAG(dag_id="example_dynamic_task_mapping", start_date=datetime(2022, 3, 4), schedule_interval='@once') as dag:

    # Warnings case: @task(map_index_template='{{ my_var }}')
    # No warnings case: @task
    @task(map_index_template='{{ my_var }}')
    def add_one(x: int):
        context = get_current_context()
        context['my_var'] = f'{x=}'
        return x + 1

    @task
    def sum_it(values):
        total = sum(values)
        print(f"Total was {total}")

    added_values = add_one.expand(x=[1, 2, 3])
    sum_it(added_values)

```

Screenshot:
<img width="957" alt="2024-10-07 17_18_14-example_dynamic_task_mapping" src="https://github.com/user-attachments/assets/742422fa-d4f5-4972-bf45-30fc79454473">

**Comment by shahar1** [comment]
Closing as I recreated the issue here: https://github.com/apache/airflow/issues/42806.
@foldvari - thanks for reproducing it, everyone is welcome to contribute a solution for this bug.
